src/aws/flb_aws_credentials_http.c

```diff
@@ -499,6 +507,10 @@ struct flb_aws_credentials *flb_parse_json_credentials(char *response,
                 t = &tokens[i];
                 current_token = &response[t->start];
                 len = t->end - t->start;
+                if (creds->session_token != NULL) {
+                    flb_error("Trying to double allocate session_token");
+                    goto error;
```

Comment:
if this is meant to support fuzzing this input, should we just continue here instead?

In a real response, setting the access key twice would mean the credential server was broken or something or that something else had gone wrong such that the response doesn't match the API spec.

Reply:
I'm happy to continue: if so we have the options of:
1) free what is in `creds-session_token`
2) just ignore leaks in this code
3) somehow try to enforce a precondition on the fuzzer to not trigger this -- although I would advice against this and rather and then rather opt in for (2)

Reply:
> In a real response, setting the access key twice would mean the credential server was broken or something or that something else had gone wrong such that the response doesn't match the API spec.

I'm not sure about the threat model of this particular code so can't speak on the attack surface as such, however, in a sense what the fuzzing is trying to simulate what is possible in the event something has gone wrong.

Reply:
fyi these checks were added because of the leaks found by CIFuzz in the above runs, e.g.:
```
Direct leak of 85 byte(s) in 5 object(s) allocated from:
    #0 0x5319d6 in malloc /src/llvm-project/compiler-rt/lib/asan/asan_malloc_linux.cpp:69:3
    #1 0x5715bc in flb_malloc /src/fluent-bit/include/fluent-bit/flb_mem.h:80:12
    #2 0x5715bc in sds_alloc /src/fluent-bit/src/flb_sds.c:41:11
    #3 0x5715bc in flb_sds_create_len /src/fluent-bit/src/flb_sds.c:62:9
    #4 0xde0eba in flb_parse_json_credentials /src/fluent-bit/src/aws/flb_aws_credentials_http.c:476:40
    #5 0xde0869 in flb_parse_http_credentials /src/fluent-bit/src/aws/flb_aws_credentials_http.c:394:12
    #6 0x56d8d9 in fuzz_http /src/fluent-bit/tests/internal/fuzzers/aws_credentials_fuzzer.c:81:13
    #7 0x56d8d9 in LLVMFuzzerTestOneInput /src/fluent-bit/tests/internal/fuzzers/aws_credentials_fuzzer.c:125:5
    #8 0x440843 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:611:15
    #9 0x44002a in fuzzer::Fuzzer::RunOne(unsigned char const*, unsigned long, bool, fuzzer::InputInfo*, bool, bool*) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:514:3
    #10 0x4416f9 in fuzzer::Fuzzer::MutateAndTestOne() /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:757:19
    #11 0x4423c5 in fuzzer::Fuzzer::Loop(std::__Fuzzer::vector<fuzzer::SizedFile, std::__Fuzzer::allocator<fuzzer::SizedFile> >&) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerLoop.cpp:895:5
    #12 0x43172f in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerDriver.cpp:912:6
    #13 0x45ad82 in main /src/llvm-project/compiler-rt/lib/fuzzer/FuzzerMain.cpp:20:10
    #14 0x7fe3086b8082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082) (BuildId: 1878e6b475720c7c51969e69ab2d276fae6d1dee)
 ```

Reply:
if we continue, then we should not allocate again, and there should be no leak right? With a continue or an error return, either way, we guarantee we only alloc each field once. 

The threat model POV is a good point though, if the response has a key mentioned more than once, that is odd and does not conform to the spec, and since this is cred handling, may be if anything goes different than we expect then we throw an error and give up immediately. 

So you can keep as is.